x-pack/plugins/encrypted_saved_objects/server/saved_objects/encrypted_saved_objects_client_wrapper.ts

```diff
@@ -235,17 +262,23 @@ export class EncryptedSavedObjectsClientWrapper implements SavedObjectsClientCon
    * Strips encrypted attributes from any bulk Saved Objects API response. If type for any bulk
    * response portion isn't registered, it is returned as is.
    * @param response Raw response returned by the underlying base client.
+   * @param [objects] Optional list of saved objects with original attributes.
+   * @param [namespace] Optional namespace that was used for the saved objects operation.
    */
-  private stripEncryptedAttributesFromBulkResponse<
-    T extends SavedObjectsBulkResponse | SavedObjectsFindResponse | SavedObjectsBulkUpdateResponse
-  >(response: T): T {
-    for (const savedObject of response.saved_objects) {
-      if (this.options.service.isRegistered(savedObject.type) && savedObject.attributes) {
-        savedObject.attributes = this.options.service.stripEncryptedAttributes(
-          savedObject.type,
-          savedObject.attributes as Record<string, unknown>
-        );
-      }
+  private async handleEncryptedAttributesInBulkResponse<
+    T,
+    R extends
+      | SavedObjectsBulkResponse<T>
+      | SavedObjectsFindResponse<T>
+      | SavedObjectsBulkUpdateResponse<T>,
+    O extends Array<SavedObjectsBulkCreateObject<T>> | Array<SavedObjectsBulkUpdateObject<T>>
+  >(response: R, objects?: O, namespace?: string) {
+    for (const [index, savedObject] of response.saved_objects.entries()) {
+      await this.handleEncryptedAttributesInResponse(
```

Comment:
**question** do we want this to be handled serially, or should we try to parallelize this?

Reply:
I'd go with sequential processing first and see if it's enough to be honest. If we decide to parallelize it we may need to have a cap on how many objects we process in parallel to not exhaust NodeJS thread pool, otherwise bulk operations with really large amount of SOs (even without encrypted attributes) may slow Kibana down as a whole. What do you think?

Reply:
++ makes perfect sense to me, I just wanted to ask in case this wasn't your intent